### PR TITLE
add error handlers pacbio flow

### PR DESCRIPTION
**Pull request by ChrOertlin**
## Description

add error handling to pacbio flow

### Added

-

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/services/post_processing/abstract_classes.py [hunk 1]
```diff
@@ -11,7 +11,7 @@ class RunDataGenerator(ABC):
 
     @abstractmethod
     def get_run_data(self, run_name: str, sequencing_dir: str) -> RunData:
-        pass
+        raise NotImplementedError
```
Comment (diitaz93) [inline cg/services/post_processing/abstract_classes.py:14]: this is unnecessary, the abstract method decorator should be used with `pass`

File: cg/services/post_processing/error_handlers.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+"""Module that holds the error handlers for the Post Processing service."""
+
+import logging
+from cg.exc import CgError
+
+LOG = logging.getLogger(__name__)
+
+
+def handle_post_processing_errors(to_except: tuple, to_raise):
```
Comment (diitaz93) [inline cg/services/post_processing/error_handlers.py:9]:
```suggestion
def handle_post_processing_errors(to_except: tuple, to_raise: Type[Exception]):
```

File: cg/services/post_processing/error_handlers.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    """Handle errors for the Post processing services."""
+
+    def error_handling(fn):
```
Comment (diitaz93) [inline cg/services/post_processing/error_handlers.py:12]: Nice! can't this be `@wraps(func)`?

**Review by diitaz93 — COMMENTED** [review]
Looking good 💯

File: cg/services/post_processing/error_handlers.py (hunk 3 above)
Comment (ChrOertlin) [reply cg/services/post_processing/error_handlers.py:12]: It complains of not getting a parameter. I will read up on the proper usage.

File: cg/services/post_processing/error_handlers.py (hunk 2 above)
Comment (ChrOertlin) [reply cg/services/post_processing/error_handlers.py:9]: It is not an Type Exception, I tried those type hints but it was complaining. I will look up waht those should be

File: tests/services/post_processing/pacbio/run_data_generator/test_pacbio_run_data_generator.py [hunk 4]
```diff
@@ -25,3 +28,33 @@ def test_get_run_data(
...
+
+    # WHEN Generating run data
+    # THEN an error is raised
+    with pytest.raises(PostProcessingRunDataGeneratorError):
+        run_data_generator.get_run_data(
+            run_name=run_name, sequencing_dir=pac_bio_runs_dir.as_posix()
+        )
+
+
+def test_get_run_data_wrong_suffix(
```
Comment (diitaz93) [inline tests/services/post_processing/pacbio/run_data_generator/test_pacbio_run_data_generator.py:48]: these can be one parametrised test

File: tests/services/post_processing/pacbio/run_file_manager/test_pacbio_run_file_manager.py [hunk 5]
```diff
@@ -41,3 +45,43 @@ def test_get_files_to_store(
     # THEN the correct files are returned
     full_list: list[Path] = pac_bio_report_files_to_parse + pac_bio_hifi_files
     assert set(files) == set(full_list)
+
+
+def test_get_files_to_store_error(
+    expected_pac_bio_run_data: PacBioRunData,
+):
+    """Test that the files to be stored are returned"""
```
Comment (diitaz93) [inline tests/services/post_processing/pacbio/run_file_manager/test_pacbio_run_file_manager.py:53]:
```suggestion
    """Test that the files to be stored are returned."""
```

File: tests/services/post_processing/pacbio/run_file_manager/test_pacbio_run_file_manager.py [hunk 6]
```diff
@@ -41,3 +45,43 @@ def test_get_files_to_store(
...
+        # WHEN getting the files to store
+        # THEN an error is raised
+        with pytest.raises(PostProcessingRunFileManagerError):
+            file_manager.get_files_to_store(expected_pac_bio_run_data)
+
+
+def test_get_files_to_parse_error(
+    expected_pac_bio_run_data: PacBioRunData,
+):
+    """Test that the files to be stored are returned"""
```
Comment (diitaz93) [inline tests/services/post_processing/pacbio/run_file_manager/test_pacbio_run_file_manager.py:73]:
```suggestion
    """Test that the files to be stored are returned."""
```

**Review by diitaz93 — APPROVED** [review]
💯 Cool to test error with mocks with side effects 👌

**Pull request merged**